### PR TITLE
Add address verification checks to routes (and Error refactor)

### PR DESCRIPTION
**Pull request by zakhap**
## Description

In routes, we frequently call `User.getAddresses()` however we hadn't previously checked to see if they're all verified. I added a verification filter `.filter((addr) => !!addr.verified)` to all these calls or in-chain with `.map()` contextually. All previous filters with the usage `.filter((addr) => addr.verified)` without the `!!` have been modified to include the `!!` as it forces the value to be a boolean. 

### Addendum:
Per @drewstone's request, I also factored out all error messages into objects at the top of each route and modified all previously written error tests to test the error type as well. No new tests were written in this PR.

Closing #162 

## Motivation and Context

Security reasons for our users. Gotta make sure they're verified!

## How has this been tested?

`yarn test-api` still passes after dozens of files were modified.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] yes, and they are tested: all pass like before, no change in coverage.

### CODE REVIEW

File: server/routes/deleteComment.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ const deleteComment = async (models, req: Request, res: Response, next: NextFunc
       where: { id: req.body.comment_id, },
       include: [ models.Address ],
     });
-    if (userOwnedAddresses.filter((addr) => addr.verified).map((addr) => addr.id).indexOf(comment.address_id) === -1) {
+    if (userOwnedAddresses.filter((addr) => !!addr.verified).map((addr) => addr.id).indexOf(comment.address_id) === -1) {
       return next(new Error('Not owned by this user'));
```
Comment (drewstone) [inline server/routes/deleteComment.ts:21]: Can you consolidate errors messages for routes that don't have them yet?

File: server/routes/deleteCommunity.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ const deleteCommunity = async (models, req: Request, res: Response, next: NextFu
     const community = await models.OffchainCommunity.findOne({
       where: { id: req.body.community_id },
     });
-    if (userOwnedAddresses.filter((addr) => addr.verified).map((addr) => addr.id).indexOf(community.creator_id) === -1) {
+    if (userOwnedAddresses.filter((addr) => !!addr.verified).map((addr) => addr.id).indexOf(community.creator_id) === -1) {
       return next(new Error('Only the original creator can delete this community'));
```
Comment (drewstone) [inline server/routes/deleteCommunity.ts:20]: Here too

File: server/routes/deleteReaction.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ const deleteReaction = async (models, req: Request, res: Response, next: NextFun
       where: { id: req.body.reaction_id, },
       include: [ models.Address ],
     });
-    if (userOwnedAddresses.filter((addr) => addr.verified).map((addr) => addr.id).indexOf(reaction.address_id) === -1) {
+    if (userOwnedAddresses.filter((addr) => !!addr.verified).map((addr) => addr.id).indexOf(reaction.address_id) === -1) {
       return next(new Error('Not owned by this user'));
```
Comment (drewstone) [inline server/routes/deleteReaction.ts:21]: Here too

File: server/routes/editComment.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ const editComment = async (models, req: Request, res: Response, next: NextFuncti
       where: { id: req.body.id },
     });
 
-    if (userOwnedAddresses.filter((addr) => addr.verified).map((addr) => addr.id).indexOf(comment.address_id) === -1) {
+    if (userOwnedAddresses.filter((addr) => !!addr.verified).map((addr) => addr.id).indexOf(comment.address_id) === -1) {
       return next(new Error('Not owned by this user'));
```
Comment (drewstone) [inline server/routes/editComment.ts:44]: Here too

**Comment by zakhap** [comment]
@drewstone I'll factor out the Error messages and put in tests for them. 

TODOs: 
- [ ] Factor out Error messages,
- [ ] Write tests for Error messages

**Comment by zakhap** [comment]
@drewstone As of now, all Errors have been factored out within all of the route files. All routes that throw errors have been attended to. Moreover, all tests already written that previously checked only if an error was thrown, are now checking the error type as well. No new tests were written in this PR, only modified to new standard where written.

**Pull request merged**